core/src/main/java/tc/oc/pgm/gamerules/GameRule.java

```diff
@@ -1,30 +1,78 @@
 package tc.oc.pgm.gamerules;
 
-public enum GameRule {
-  DO_FIRE_TICK("doFireTick"),
-  DO_MOB_LOOT("doMobLoot"),
-  DO_TILE_DROPS("doTileDrops"),
-  MOB_GRIEFING("mobGriefing"),
-  NATURAL_REGENERATION("naturalRegeneration"),
-  DO_DAYLIGHT_CYCLE("doDaylightCycle");
-
-  private String value;
-
-  GameRule(String value) {
-    this.value = value;
+import static com.google.common.base.Preconditions.*;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Predicate;
+
+public class GameRule<T> {
+  private static final Map<String, GameRule<?>> RULES_BY_NAME = new HashMap<>();
+
+  public static final GameRule<Boolean> DO_DAYLIGHT_CYCLE =
+      register(new BooleanGameRule("doDaylightCycle"));
+  public static final GameRule<Boolean> DO_ENTITY_DROPS =
+      register(new BooleanGameRule("doEntityDrops"));
+  public static final GameRule<Boolean> DO_FIRE_TICK = register(new BooleanGameRule("doFireTick"));
+  public static final GameRule<Boolean> DO_MOB_LOOT = register(new BooleanGameRule("doMobLoot"));
+  public static final GameRule<Boolean> DO_MOB_SPAWNING =
+      register(new BooleanGameRule("doMobSpawning"));
+  public static final GameRule<Boolean> DO_TILE_DROPS =
+      register(new BooleanGameRule("doTileDrops"));
+  public static final GameRule<Boolean> KEEP_INVENTORY =
+      register(new BooleanGameRule("keepInventory"));
+  public static final GameRule<Boolean> MOB_GRIEFING = register(new BooleanGameRule("mobGriefing"));
+  public static final GameRule<Boolean> NATURAL_REGENERATION =
+      register(new BooleanGameRule("naturalRegeneration"));
+  public static final GameRule<Integer> RANDOM_TICK_SPEED =
+      register(new IntegerGameRule("randomTickSpeed"));
+  public static final GameRule<Boolean> REDUCED_DEBUG_INFO =
+      register(new BooleanGameRule("reducedDebugInfo"));
```

Comment:
I'd rather not maintain a list of game rules, just allow for any game rule or any value.